HW/baseball.sql

```diff
@@ -0,0 +1,29 @@
+
+-- first join the game and batter counts table and save the results as a temporary table
+drop temporary table if exists joined_batter_game;
+create temporary table joined_batter_game
+select game.local_date,batter_counts.batter,batter_counts.game_id,batter_counts.Hit,batter_counts.atBat from batter_counts join game on batter_counts.game_id=game.game_id;
+-- change batter data type to character because it was integer
+alter table joined_batter_game modify batter varchar(100);
+-- calculate annual batting average
+drop temporary table if exists annual_at_bat_per_player;
+create temporary table annual_at_bat_per_player
+select batter,year(local_date), sum(Hit)/sum(nullif(atBat,0)) as Average_at_bat from joined_batter_game group by batter,year(local_date) order by year(local_date);
```

Comment:
I ran this for a specific batter and it looks right. I saw some batters with 1.00 averages but that might just be that they didn't have many at-bats.